frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-thick-provisioner.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { ProvisionerProps } from '@console/plugin-sdk';
+import { useFlag } from '@console/shared/src/hooks/flag';
+
+import { GUARDED_FEATURES } from '../../features';
+
+import './ocs-storage-class-form.scss';
+
+export const ThickProvision: React.FC<ProvisionerProps> = ({ onParamChange }) => {
+  const { t } = useTranslation();
+  const isThickProvisionSupported = useFlag(GUARDED_FEATURES.THICK_PROVISION);
+
+  const [checked, isChecked] = React.useState(false);
+
+  const setChecked = (value: boolean) => {
+    onParamChange(value.toString());
+    isChecked(value);
+  };
+
+  return (
+    isThickProvisionSupported && (
+      <>
+        <div className="checkbox ocs-storage-class-thickprovision__checkbox">
+          <label>
+            <input
+              type="checkbox"
+              onChange={(event) => setChecked(event.target.checked)}
+              checked={checked}
+            />
+            {t('ceph-storage-plugin~Enable Thick Provisioning.')}
+          </label>
+        </div>
+        <span className="help-block">
+          {t(
+            'ceph-storage-plugin~By enabling thick-provisioning, volumes will allocate the requested capacity upon volume creation. Volume creation to be slower when thick-provisioning is enabled.',
+          )}
+        </span>
```

Comment:
I can use <Checkbox> here but it makesense to use when we want to display any badge:
for example:
![image](https://user-images.githubusercontent.com/15073590/111221916-2217b380-8601-11eb-8a57-b83c463fed96.png)

Without a badge, its alignment is not looking good